core/src/test/java/org/apache/iceberg/TestFilesTableTaskParser.java

```diff
@@ -0,0 +1,131 @@
+    assertThatThrownBy(() -> FilesTableTaskParser.fromJson(rootNode.get("arr")))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining("Invalid JSON node for files task: non-object ");
+  }
+
+  @Test
+  public void testParser() {
+    BaseFilesTable.ManifestReadTask task = createTask();
+    String jsonStr = ScanTaskParser.toJson(task);
+    assertThat(jsonStr).isEqualTo(taskJson());
+    BaseFilesTable.ManifestReadTask deserializedTask =
+        (BaseFilesTable.ManifestReadTask) ScanTaskParser.fromJson(jsonStr, false);
+    assertTaskEquals(task, deserializedTask);
+  }
+
+  static BaseFilesTable.ManifestReadTask createTask() {
+    Schema schema = TestBase.SCHEMA;
+    HadoopFileIO fileIO = new HadoopFileIO();
+    fileIO.initialize(ImmutableMap.of("k1", "v1", "k2", "v2"));
+    Map<Integer, PartitionSpec> specsById =
+        PartitionUtil.indexSpecs(
+            Arrays.asList(PartitionSpec.builderFor(schema).bucket("data", 16).build()));
+    ManifestFile manifestFile = TestManifestFileParser.createManifestFile();
+    return new BaseFilesTable.ManifestReadTask(
+        schema, fileIO, specsById, manifestFile, schema, Expressions.equal("id", 1));
+  }
+
+  private String taskJson() {
+    return "{\"task-type\":\"files-table-task\","
+        + "\"schema\":{\"type\":\"struct\",\"schema-id\":0,\"fields\":[{"
+        + "\"id\":3,\"name\":\"id\",\"required\":true,\"type\":\"int\"},"
+        + "{\"id\":4,\"name\":\"data\",\"required\":true,\"type\":\"string\"}]},"
+        + "\"projection\":{\"type\":\"struct\",\"schema-id\":0,\"fields\":[{"
+        + "\"id\":3,\"name\":\"id\",\"required\":true,\"type\":\"int\"},"
+        + "{\"id\":4,\"name\":\"data\",\"required\":true,\"type\":\"string\"}]},"
+        + "\"file-io\":{\"io-impl\":\"org.apache.iceberg.hadoop.HadoopFileIO\","
+        + "\"properties\":{\"k1\":\"v1\",\"k2\":\"v2\"}},"
+        + "\"partition-specs\":[{\"spec-id\":0,\"fields\":[{"
+        + "\"name\":\"data_bucket\",\"transform\":\"bucket[16]\",\"source-id\":4,\"field-id\":1000}]}],"
+        + "\"residual-filter\":{\"type\":\"eq\",\"term\":\"id\",\"value\":1},"
+        + "\"manifest-file\":{\"path\":\"/path/input.m0.avro\","
+        + "\"length\":5878,\"partition-spec-id\":0,\"content\":0,\"sequence-number\":1,\"min-sequence-number\":2,"
+        + "\"added-snapshot-id\":12345678901234567,"
+        + "\"added-files-count\":1,\"existing-files-count\":3,\"deleted-files-count\":0,"
+        + "\"added-rows-count\":10,\"existing-rows-count\":30,\"deleted-rows-count\":0,"
+        + "\"partition-field-summary\":[{\"contains-null\":true,\"contains-nan\":false,"
+        + "\"lower-bound\":\"0A000000\",\"upper-bound\":\"64000000\"}],\"key-metadata\":\"DB030000\"}}";
+  }
+
+  private void assertTaskEquals(
+      BaseFilesTable.ManifestReadTask expected, BaseFilesTable.ManifestReadTask actual) {
+    assertThat(expected.schema().sameSchema(actual.schema())).as("Schema should match").isTrue();
```

Comment:
`assertThat(actual.schema().asStruct()).isEqualTo(expected.schema().asStruct());` would be slightly better, because you would immediately see the differences in the schema when the assertion fails. Using `isTrue` would require debugging both schemas to understand the difference.
Could you please update this also for the projection check in the next few lines and also do the same in `TestAllManifestsTableTaskParser`?